test/widgets/set_status_test.dart

```diff
@@ -0,0 +1,552 @@
+        });
+
+        testWidgets('emoji & text are changed, then reset -> buttons are enabled, then "Clear" is enabled, "Save" is not', (tester) async {
+          prepareBoringImageHttpClient();
+
+          Route<dynamic>? currentRoute;
+          final testNavObserver = TestNavigatorObserver();
+          testNavObserver.onPushed = (route, _) => currentRoute = route;
+          testNavObserver.onPopped = (_, previous) => currentRoute = previous;
+
+          await setupPage(tester,
+            emojiData: serverEmojiDataPopularPlus(suggestedEmojiData),
+            navigatorObserver: testNavObserver,
+            change: UserStatusChange(
+              text: OptionSome('Happy'),
+              emoji: OptionSome(StatusEmoji(emojiName: 'slight_smile',
+                emojiCode: '1f642', reactionType: ReactionType.unicodeEmoji))));
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: true);
+          checkButtonEnabled(tester, saveButtonFinder, expected: false);
+
+          // Choose 'calender' included in suggested emojis.
+          await chooseEmojiFromPicker(tester, '1f4c5',
+            emojiSelected: true,
+            navObserver: testNavObserver, currentRoute: () => currentRoute!);
+          await tester.enterText(findStatusTextField(), 'Happy as a calm');
+          await tester.pump();
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: true);
+          checkButtonEnabled(tester, saveButtonFinder, expected: true);
+
+          // Reset to the initial emoji.
+          await chooseEmojiFromPicker(tester, '1f642',
+            emojiSelected: true,
+            navObserver: testNavObserver, currentRoute: () => currentRoute!);
+          // Reset to the initial text.
+          await tester.enterText(findStatusTextField(), 'Happy');
+          await tester.pump();
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: true);
+          checkButtonEnabled(tester, saveButtonFinder, expected: false);
+
+          debugNetworkImageHttpClientProvider = null;
+        });
+
+        testWidgets('a new suggestion is selected, then reset -> buttons are enabled, then "Clear" is enabled, "Save" is not', (tester) async {
+          await setupPage(tester,
+            emojiData: suggestedEmojiData,
+            // One of the emoji suggestions.
+            change: UserStatusChange(
+              text: OptionSome('Working remotely'),
+              emoji: OptionSome(StatusEmoji(emojiName: 'house',
+                emojiCode: '1f3e0', reactionType: ReactionType.unicodeEmoji))));
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: true);
+          checkButtonEnabled(tester, saveButtonFinder, expected: false);
+
+          await tester.tap(findSuggestion(code: '1f6e0', text: 'Busy'));
+          await tester.pump();
+
+          checkButtonEnabled(tester, clearButtonFinder, expected: true);
+          checkButtonEnabled(tester, saveButtonFinder, expected: true);
```

Comment:
These two helpers almost always get called in pairs, for the clear and save buttons. How about combining those into one helper?

Then those call sites can generally all be joined to the stanzas above them, following the usual pattern I've mentioned before where a round of "set up, then check" is one stanza. (In fact that'd be a good change even with the two separate helper calls, but especially so with just one.) So:
```suggestion
          await tester.pump();
          checkButtonsEnabled(tester, expectClear: true, expectSave: true);
```